test/e2e/realtimecompositions_test.go

```diff
@@ -0,0 +1,111 @@
+	// label to be assigned to tests that should be part of the Usage test
+	// suite.
+	SuiteRealtimeCompositions = "realtime-compositions"
+)
+
+func init() {
+	environment.AddTestSuite(SuiteRealtimeCompositions,
+		config.WithHelmInstallOpts(
+			helm.WithArgs("--set args={--debug,--enable-realtime-compositions}"),
+		),
+		config.WithLabelsToSelect(features.Labels{
+			config.LabelTestSuite: []string{SuiteRealtimeCompositions, config.TestSuiteDefault},
+		}),
+	)
+}
+
+// TestRealtimeCompositions tests scenarios for compositions with realtime
+// reconciles through MR updates.
+func TestRealtimeCompositions(t *testing.T) {
+	manifests := "test/e2e/manifests/apiextensions/composition/realtime-compositions"
+
+	nopCrossplaneList := composed.NewList(composed.FromReferenceToList(corev1.ObjectReference{
+		APIVersion: "nop.crossplane.io/v1alpha1",
+		Kind:       "NopResource",
+	}))
+	nopList := composed.NewList(composed.FromReferenceToList(corev1.ObjectReference{
+		APIVersion: "realtime-revision-selection.e2e.crossplane.io/v1alpha1",
+		Kind:       "NopResource",
+	}))
+	xnopList := composed.NewList(composed.FromReferenceToList(corev1.ObjectReference{
+		APIVersion: "realtime-revision-selection.e2e.crossplane.io/v1alpha1",
+		Kind:       "XNopResource",
+	}))
+	withTestLabels := resources.WithLabelSelector(labels.FormatLabels(map[string]string{"realtime-compositions": "true"}))
+
+	environment.Test(t,
+		features.New(t.Name()).
+			WithLabel(LabelStage, LabelStageAlpha).
+			WithLabel(LabelArea, LabelAreaAPIExtensions).
+			WithLabel(LabelSize, LabelSizeSmall).
+			WithLabel(LabelModifyCrossplaneInstallation, LabelModifyCrossplaneInstallationTrue).
+			WithLabel(config.LabelTestSuite, SuiteRealtimeCompositions).
+			WithSetup("EnableAlphaRealtimeCompositions", funcs.AllOf(
+				funcs.AsFeaturesFunc(environment.HelmUpgradeCrossplaneToSuite(SuiteRealtimeCompositions)),
+				funcs.ReadyToTestWithin(1*time.Minute, namespace),
+			)).
+			WithSetup("PrerequisitesAreCreated", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "setup/*.yaml"),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "setup/*.yaml"),
+				funcs.ResourcesHaveConditionWithin(1*time.Minute, manifests, "setup/definition.yaml", apiextensionsv1.WatchingComposite()),
+				funcs.ResourcesHaveConditionWithin(2*time.Minute, manifests, "setup/provider.yaml", pkgv1.Healthy(), pkgv1.Active()),
+			)).
+			Assess("CreateClaim", funcs.AllOf(
+				funcs.ApplyResources(FieldManager, manifests, "claim.yaml"),
+				funcs.InBackground(funcs.LogResources(nopList, withTestLabels)),
+				funcs.InBackground(funcs.LogResources(xnopList, withTestLabels)),
+				funcs.InBackground(funcs.LogResources(nopCrossplaneList, withTestLabels)),
+				funcs.ResourcesCreatedWithin(30*time.Second, manifests, "claim.yaml"),
+				funcs.ResourcesHaveConditionWithin(5*time.Minute, manifests, "claim.yaml", xpv1.Available()),
+			)).
+			Assess("UpdateMR", funcs.AllOf(
```

Comment:
I think we can even avoid this step, because nop provider updates MRs even above, so line 83 would work even under a way shorter timeout.

Reply:
Don't think. I had to switch the nop manifest to immediate readiness, because it is unexpectedly slow to do so if you e.g. set it to `1s`. So now it is `0s`. And as a composition does reconciles frequently in the beginning for different reasons, I don't want to step into that noise. Hence, this step is explicitly changing something *after* readiness, when the noise is hopefully over.